sdks/java/io/xml/src/test/java/org/apache/beam/sdk/io/xml/XmlSourceTest.java

```diff
@@ -526,13 +560,21 @@ public void testReadXMLInvalidRecordClass() throws IOException {
     File file = tempFolder.newFile("trainXMLSmall");
     Files.write(file.toPath(), trainXML.getBytes(StandardCharsets.UTF_8));
```

Comment:
I don't believe this shows that the validation handler is being used since after updating the test, our test expectations haven't changed.

Also, I believe the existing test is valid and we should have a new test which shows that the validation handler is being invoked.